python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -718,6 +720,23 @@ def _validate_input(self):
                     raise NotImplementedError(
                         "Regex contains only supports a scalar pattern"
                     )
+        elif self.name == pl_expr.StringFunction.Replace:
+            _, literal = self.options
+            if not literal:
+                raise NotImplementedError("literal=False is not supported for replace")
+            if not all(isinstance(expr, Literal) for expr in self.children[1:]):
+                raise NotImplementedError("replace only supports scalar target")
+            target = self.children[1]
+            if target.value == pa.scalar("", type=pa.string()):
+                raise NotImplementedError(
+                    "libcudf replace does not support empty strings"
```

Comment:
I think the conclusion of yesterdays discussion was that it's probably not worth supporting this one since polars was doing something where it's replacing after every character (instead of it being a no-op).

Might have remembered wrong, though.

Reply:
yeah, let's raise for now